### PR TITLE
feat: add exchange rate and payment utxos methods to provider

### LINKED ISSUE #71: Add getPaymentUtxos() method to the provider

### LINKED ISSUE #82: Add getExchangeRate method to provider with cache

### PR DESCRIPTION
**Pull request by danwag06**

### CODE REVIEW

File: public/background.js [hunk 1]
```diff
@@ -271,6 +277,104 @@ const processGetOrdinalsRequest = (sendResponse) => {
...
+    );
+    if (!res.ok) {
+      throw new Error(`Fetch error: ${res.status} - ${res.statusText}`);
+    }
+
+    const items = await res.json();
+
+    let utxoData = [];
+    for (const i of items) {
+      const scriptRes = await fetch(
```
Comment (shruggr) [inline public/background.js]: I was misunderstanding how this was going to work, and that an address was going to be passed in. This makes total sense, but In that case, we really don't need script, and we shouldn't look it up. I was expecting more magic, but this lookup is not necessary as the calling party can generate the script themselves.

**Review by shruggr — CHANGES_REQUESTED** [review]

File: public/background.js [hunk 2]
```diff
@@ -271,6 +277,104 @@ const processGetOrdinalsRequest = (sendResponse) => {
...
+    });
+    return;
+  }
+  try {
+    const requestOptions = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ addresses: message.params.addresses }),
```
Comment (shruggr) [inline public/background.js]: I wasn't picturing you could request UTXOs for any address. I was picturing the wallet delivering this based on your payment address, and we'll deliver your utxos from local storage, once we have them there, but for now we just use WoC behind the scenes

File: public/background.js (hunk 1 above)
Comment (danwag06) [reply public/background.js]: Makes sense. So it really should just be getPaymentUtxos() with no params. Will cache as well and update when the user launches the app.

**Review by shruggr — COMMENTED** [review]
I think I don't understand `background.js`, where it runs, and how it functions, but I think this works.

**Review by shruggr — APPROVED** [review]

**Pull request merged**